src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -0,0 +1,974 @@
+            name_lit = Literal(str(signature), CHARACTER_TYPE)
+            LFRicExtractDriverCreator.add_call(program, read_var,
+                                               [name_lit, Reference(sym)])
+
+        # Then handle all variables that are written (note that some
+        # variables might be read and written)
+        # ----------------------------------------------------------
+        # Collect all output symbols to later create the tests for
+        # correctness. This list stores 2-tuples: first one the
+        # variable that stores the output from the kernel, the second
+        # one the variable that stores the output values read from the
+        # file. The content of these two variables should be identical
+        # at the end.
+        output_symbols = []
+        for signature in output_list:
+            # Find the right symbol for the variable. Note that all variables
+            # in the input and output list have been detected as being used
+            # when the variable accesses were analysed. Therefore, these
+            # variables have References, and will already have been declared
+            # in the symbol table (in add_all_kernel_symbols).
+            orig_sym = original_symbol_table.lookup(signature[0])
+            is_input = signature in input_list
+            if orig_sym.is_array and orig_sym.datatype.intrinsic.name in \
+                    self._all_field_types:
+                flattened = LFRicExtractDriverCreator.\
+                    flatten_string(str(signature))
+                upper = int(orig_sym.datatype.shape[0].upper.value)
+                for i in range(1, upper+1):
+                    self.create_output_var_code(flattened, program, is_input,
+                                                read_var, postfix,
+                                                output_symbols, index=i)
+            else:
+                sig_str = str(signature)
+                self.create_output_var_code(sig_str, program, is_input,
+                                            read_var, postfix,
+                                            output_symbols)
+        return output_symbols
+
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def import_modules(program, sched):
+        '''This function adds all the import statements required for the
+        actual kernel calls. It finds all calls in the PSyIR tree and
+        checks for calls with a ImportInterface. Any such call will
+        get a ContainerSymbol added for the module, and a RoutineSymbol
+        with an import interface pointing to this module.
+
+        :param program: the PSyIR Routine to which any code must \
+            be added. It also contains the symbol table to be used.
+        :type program: :py:class:`psyclone.psyir.nodes.Routine`
+        :param sched: the schedule that will be called by the driver \
```

Comment:
This confused me (maybe I'm having a slow patch). I would have expected that the "schedule that will be called by the driver" to *be* the program (Routine). Is it because we haven't copied it over yet?

Reply:
The (driver) program calls the kernel directly, so the schedule is in the driver. So the program (Routine) has the schedule as child, and the kernel invocations inside of the schedule will be called.

Reply:
OK, so maybe "the schedule to analyse for module imports"? Is `program` going to be the driver routine? If so, please could you extend the description at L605 to make this clear.

Reply:
I've updated the comments, but actually also changed the parameter - instead of 'program' it takes a symbol table (that's the only reason program was passed in), and I added comments when calling this function on what is happening. I hope that is clearer now.

Reply:
Yes, that's good, thanks.